src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs

```diff
@@ -0,0 +1,618 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    /// <summary>
+    /// Base class for <see cref="IProjectDependenciesSubTreeProvider"/> implementations that need to 
+    /// subscribe to design time build to get data.
+    /// </summary>
+    internal class DependenciesSubTreeProviderBase : OnceInitializedOnceDisposed, IProjectDependenciesSubTreeProvider
+    {
+        /// <summary>
+        /// Gets the project asynchronous tasks service.
+        /// </summary>
+        [Import(ExportContractNames.Scopes.UnconfiguredProject)]
+        protected IProjectAsynchronousTasksService UnconfiguredProjectAsynchronousTasksService { get; private set; }
+
+        /// <summary>
+        /// Gets the subscription service for source items.
+        /// </summary>
+        [Import]
```

Comment:
i would not do that here - it is a base class and i don't want all it's implementers to have to remember to add [Import(ExportContractNames.Scopes.UnconfiguredProject)] attribute in their Importing constructors ....

---
In reply to: [74458158](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74458158) [](ancestors = 74458158)